lib/broadcast-operator.ts

```diff
@@ -280,6 +283,28 @@ export class RemoteSocket<EmitEvents extends EventsMap>
     return this.operator.emit(ev, ...args);
   }
 
+  /**
+   * Targets a room when broadcasting.
+   *
+   * @param room
+   * @return self
+   * @public
+   */
+  public to(room: Room | Room[]): BroadcastOperator<EmitEvents> {
```

Comment:
How about creating another operator

```diff
- this.operator = new BroadcastOperator(adapter, new Set([this.id]));
+ this.selfOperator = new BroadcastOperator(adapter, new Set([this.id]));
+ this.broadcastOperator = new BroadcastOperator(adapter, new Set(), new Set([this.id]));
```

`RemoteSocket.to(...)` would simply become `this.broadcastOperator.to(...)`

What do you think?

Note: while we're at it, how about adding the `local`, `broadcast` and `except()` modifiers?

Reply:
@darrachequesne Would extending the Socket class be a better idea?

Reply:
@rennokki yes, why not! Wouldn't the Socket class extend the RemoteSocket class though?

Reply:
@darrachequesne Shouldn't RemoteSocket be a Socket with extra capabilities?

Reply:
I'm not sure, since `RemoteSocket` is not an EventEmitter (no `socket.on(...)`) and is missing the `request` attribute, for example.

But please update the PR with what you have in mind, I might be missing something.